### ISSUE TITLE
Failure on first helm release leaves Garden in bad state

### ISSUE THREAD

**Issue by Eronarn** [garden-io/garden#1312]
### Current Behavior

If you fail to deploy a Helm chart the first time, then Garden gets confused:

```
[eronarn@ip-192-168-10-243 data-depot]$ garden deploy --log-level=3
Deploy 🚀 

ℹ providers                 → Getting status...
Resource(s) PersistentVolumeClaim/garden-sync-garden-system-nfs-v2 missing from cluster
All resources missing from cluster
kubectl diff indicates all resources match the deployed resources.
Comparing expected and deployed resources...
Resource garden-docker-data is not a superset of deployed resource
Comparing expected and deployed resources...
Resource garden-docker-registry is not a superset of deployed resource
ℹ providers                 → Preparing environment...
   ℹ kubernetes                → Configuring...
   ✔ kubernetes                → Configuring... → Ready
✔ providers                 → Preparing environment... → Done
————————————————————————————————————————————————————————————————————————————————
Following modules are linked to a local path:
  data-depot-metadata-chart linked to path /Users/eronarn/Code/data-depot-db-service
————————————————————————————————————————————————————————————————————————————————
ℹ data-depot-metadata-image → Preparing build (89 files)...
ℹ data-depot-metadata-image → Getting build status for v-0f44f4f0eb...
✔ data-depot-metadata-image → Getting build status for v-0f44f4f0eb... → Done (took 1.6 sec)
ℹ data-depot-metadata-chart → Preparing build (3 files)...
ℹ data-depot-metadata-chart → Getting build status for v-201a032ae1...
ℹ data-depot-metadata-chart → Building version v-201a032ae1...
✔ data-depot-metadata-chart → Building version v-201a032ae1... → Done (took 0 sec)
ℹ data-depot-metadata-chart → Deploying version v-201a032ae1...
✖ data-depot-metadata-chart → Deploying version v-201a032ae1...

Failed deploying service 'data-depot-metadata-chart' (from module 'data-depot-metadata-chart'). Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Command failed with exit code 1 (EPERM): /Users/eronarn/.garden/tools/helm/1c6ed8021f49d4ce/darwin-amd64/helm --tiller-namespace staging --kube-context stg-admin@staging upgrade eronarn-data-depot-data-depot-metadata /Users/eronarn/Code/garden-stacks/data-depot/.garden/build/data-depot-metadata-chart/helm-chart/data-depot-metadata --install --namespace staging --timeout 30 --values /Users/eronarn/Code/garden-stacks/data-depot/.garden/build/data-depot-metadata-chart/helm-chart/data-depot-metadata/garden-values.yml

UPGRADE FAILED
Error: "eronarn-data-depot-data-depot-metadata" has no deployed releases
Error: UPGRADE FAILED: "eronarn-data-depot-data-depot-metadata" has no deployed releases
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

1 deploy task(s) failed!

See error.log for detailed error message
```
```
[eronarn@ip-192-168-10-243 data-depot]$ garden delete service data-depot-metadata-helm
ℹ providers                 → Getting status...
ℹ providers                 → Preparing environment...
   ℹ kubernetes                → Configuring...
   ✔ kubernetes                → Configuring... → Ready
✔ providers                 → Preparing environment... → Done
Could not find service(s): data-depot-metadata-helm

See error.log for detailed error message
```

However Helm does understand the failure (note the _release name_ is the same as the one where no deployed releases were found, even though it's different from the _service name_:

```
[eronarn@ip-192-168-10-243 data-depot]$ XXX/helm --tiller-namespace staging --kube-context XXX delete --purge eronarn-data-depot-data-depot-metadata
release "eronarn-data-depot-data-depot-metadata" deleted
```

### Expected behavior

If a Helm release fails, Garden either purges it, or can purge the failure on next run rather than failing with "has no deployed releases"

### Your environment

```
0.10.13
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-06T01:44:30Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.2", GitCommit:"f6278300bebbb750328ac16ee6dd3aa7d3549568", GitTreeState:"clean", BuildDate:"2019-08-05T09:15:22Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Client:
 Version:           18.06.0-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        0ffa825
 Built:             Wed Jul 18 19:05:26 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.0-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       0ffa825
  Built:            Wed Jul 18 19:13:46 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Comment by edvald** [comment]
It appears in your scenario, the release did already exist in the cluster. Otherwise we use `helm install` and not `helm upgrade`. We could however explicitly catch this scenario, where a release exists but has no deployed version (which is actually a thing I _hate_ about Helm, but that's neither here nor there...).

**Comment by edvald** [comment]
I believe this has been fixed in v0.11. Closing, but please re-open if you run into it again!
